subscription/errors.go

```diff
@@ -22,6 +22,15 @@ func (e ErrSubscriptionAlreadyExists) Error() string {
 	return fmt.Sprintf("Subscription %q already exists.", e.ID)
 }
 
+// ErrInvalidSubscriptionUpdate occurs when a client tries to update a subscription in a way that changes the subscription ID.
+type ErrInvalidSubscriptionUpdate struct {
+	ID ID
+}
+
+func (e ErrInvalidSubscriptionUpdate) Error() string {
+	return fmt.Sprintf("Invalid update. This update would change the SubscriptionID for %q.", e.ID)
```

Comment:
I'm not sure if that's important to user. The more important would be "Path of existing subscription cannot be update." or something.